### ISSUE TITLE
virt-launcher fails due to an SELinux denial on /dev/null

### ISSUE THREAD

**Issue by tiraboschi** [kubevirt/kubevirt#9434]
**What happened**:
Executing Kubevirt v0.58.1 on Kubevirtci quay.io/kubevirtci/k8s-1.24:2303092154-4b9b02c I'm not able to start any VM.

on virt-launcher pod logs I see:
```
{"component":"virt-launcher-monitor","level":"error","msg":"failed to run virt-launcher","pos":"virt-launcher-monitor.go:108","reason":"open /dev/null: permission denied","timestamp":"2023-03-13T08:44:56.159607Z"}
{"component":"virt-launcher-monitor","level":"error","msg":"monitoring virt-launcher failed","pos":"virt-launcher-monitor.go:90","reason":"open /dev/null: permission denied","timestamp":"2023-03-13T08:44:56.497455Z"}
```

If I connect to the node and check SELinux avc I see:
```
stirabos@t14s:~/go/src/github.com/kubevirt/hyperconverged-cluster-operator$ cluster/ssh.sh node01
+ source ./cluster/kubevirtci.sh
++ export KUBEVIRT_PROVIDER=k8s-1.24
++ KUBEVIRT_PROVIDER=k8s-1.24
+++ curl -L -Ss https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirtci/latest
++ export KUBEVIRTCI_TAG=2303092154-4b9b02c
++ KUBEVIRTCI_TAG=2303092154-4b9b02c
++ KUBEVIRTCI_PATH=/home/stirabos/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci
++ KUBEVIRTCI_REPO=https://github.com/kubevirt/kubevirtci.git
+ kubevirtci::install
+ '[' -d /home/stirabos/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci ']'
++ cluster::_get_repo
++ git --git-dir /home/stirabos/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci/.git remote get-url origin
+ '[' https://github.com/kubevirt/kubevirtci.git '!=' https://github.com/kubevirt/kubevirtci.git ']'
++ cluster::_get_tag
++ git -C /home/stirabos/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci describe --tags
+ '[' 2303092154-4b9b02c '!=' 2303092154-4b9b02c ']'
+ '[' '!' -d /home/stirabos/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci ']'
++ kubevirtci::path
++ echo -n /home/stirabos/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci
+ /home/stirabos/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci/cluster-up/ssh.sh node01
selecting docker as container runtime
Last login: Tue Mar 14 15:56:22 2023 from 192.168.66.2
[vagrant@node01 ~]$ sudo ausearch -m avc
----
time->Tue Mar 14 15:44:22 2023
type=PROCTITLE msg=audit(1678808662.742:748): proctitle=2F7362696E2F73657466696C6573002D71002D63002F7661722F6C69622F73656C696E75782F66696E616C2F74617267657465642F706F6C6963792F706F6C6963792E3331002F7661722F6C69622F73656C696E75782F66696E616C2F74617267657465642F636F6E74657874732F66696C65732F66696C655F636F6E746578
type=EXECVE msg=audit(1678808662.742:748): argc=5 a0="/sbin/setfiles" a1="-q" a2="-c" a3="/var/lib/selinux/final/targeted/policy/policy.31" a4="/var/lib/selinux/final/targeted/contexts/files/file_contexts"
type=SYSCALL msg=audit(1678808662.742:748): arch=c000003e syscall=59 success=yes exit=0 a0=564da57761a0 a1=564db13a2960 a2=0 a3=1a4 items=0 ppid=14092 pid=14578 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="setfiles" exe="/usr/sbin/setfiles" subj=system_u:system_r:setfiles_t:s0 key=(null)
type=AVC msg=audit(1678808662.742:748): avc:  denied  { read } for  pid=14578 comm="setfiles" path="/dev/null" dev="tmpfs" ino=181173 scontext=system_u:system_r:setfiles_t:s0 tcontext=system_u:object_r:container_runtime_tmpfs_t:s0 tclass=chr_file permissive=0
----
time->Tue Mar 14 15:44:23 2023
type=PROCTITLE msg=audit(1678808663.176:749): proctitle="/sbin/load_policy"
type=EXECVE msg=audit(1678808663.176:749): argc=1 a0="/sbin/load_policy"
type=SYSCALL msg=audit(1678808663.176:749): arch=c000003e syscall=59 success=yes exit=0 a0=564da5775c60 a1=564db1798f50 a2=0 a3=1a4 items=0 ppid=14092 pid=14582 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="load_policy" exe="/usr/sbin/load_policy" subj=system_u:system_r:load_policy_t:s0 key=(null)
type=AVC msg=audit(1678808663.176:749): avc:  denied  { write } for  pid=14582 comm="load_policy" path="pipe:[185641]" dev="pipefs" ino=185641 scontext=system_u:system_r:load_policy_t:s0 tcontext=system_u:system_r:spc_t:s0 tclass=fifo_file permissive=0
type=AVC msg=audit(1678808663.176:749): avc:  denied  { write } for  pid=14582 comm="load_policy" path="pipe:[185641]" dev="pipefs" ino=185641 scontext=system_u:system_r:load_policy_t:s0 tcontext=system_u:system_r:spc_t:s0 tclass=fifo_file permissive=0
type=AVC msg=audit(1678808663.176:749): avc:  denied  { read } for  pid=14582 comm="load_policy" path="/dev/null" dev="tmpfs" ino=181173 scontext=system_u:system_r:load_policy_t:s0 tcontext=system_u:object_r:container_runtime_tmpfs_t:s0 tclass=chr_file permissive=0
----
time->Tue Mar 14 15:50:48 2023
type=PROCTITLE msg=audit(1678809048.553:766): proctitle=2F7573722F62696E2F766972742D6C61756E636865722D6D6F6E69746F72002D2D71656D752D74696D656F75740032373673002D2D6E616D650074657374766D692D6E6B763764002D2D7569640034633163346435622D326238392D343435332D383631642D313331343033303562303030002D2D6E616D657370616365006B
type=SYSCALL msg=audit(1678809048.553:766): arch=c000003e syscall=257 success=no exit=-13 a0=ffffffffffffff9c a1=c0000397a0 a2=80000 a3=0 items=0 ppid=19192 pid=19203 auid=4294967295 uid=107 gid=107 euid=107 suid=107 fsuid=107 egid=107 sgid=107 fsgid=107 tty=(none) ses=4294967295 comm="virt-launcher-m" exe="/usr/bin/virt-launcher-monitor" subj=system_u:system_r:virt_launcher.process:s0:c192,c492 key=(null)
type=AVC msg=audit(1678809048.553:766): avc:  denied  { open } for  pid=19203 comm="virt-launcher-m" path="/dev/null" dev="tmpfs" ino=229423 scontext=system_u:system_r:virt_launcher.process:s0:c192,c492 tcontext=system_u:object_r:container_file_t:s0:c192,c492 tclass=chr_file permissive=0
[vagrant@node01 ~]$
```

**What you expected to happen**:
I'm able to start a VM

**How to reproduce it (as minimally and precisely as possible)**:
Deploy Kubevirt v0.58.1 from HCO 1.8.0-unstable on Kubevirtci quay.io/kubevirtci/k8s-1.24:2303092154-4b9b02c and try to start a VM

**Additional context**:
Not reproducible with Kubevirt v0.59.0

**Environment**:
- KubeVirt version (use `virtctl version`): Kubevirt v0.58.1
- Kubernetes version (use `kubectl version`): Kubevirtci quay.io/kubevirtci/k8s-1.24:2303092154-4b9b02c 
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

**Comment by xpivarc** [comment]
Ok, there was a time when this was broken on the KubevirtCI. It should be fixed by now, let me check.

**Comment by tiraboschi** [comment]
2303092154-4b9b02c is the latest available on https://quay.io/repository/kubevirtci/k8s-1.24?tab=tags&tag=latest

**Comment by xpivarc** [comment]
FYI @jean-edouard

**Comment by EdDev** [comment]
@tiraboschi , was this resolved?

**Comment by xpivarc** [comment]
/close 
This should be fixed in container-selinux-2.205.0
